### PR TITLE
Vue validations

### PR DESCRIPTION
**Pull request by rlajous**
## Summary

- Added validations to password and email

## Screenshot 
vuelidations

## Trello Card
https://trello.com/c/UzqcviSZ/27-form-validation

### CODE REVIEW

File: wolox-books/src/views/Register.vue [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  },
+  methods: {
+    onSubmit() {
+      const data = {
+        first_name: this.firstName,
+        last_name: this.lastName,
+        email: this.email,
+        password: this.password
+      };
+      console.log(data);
```
Comment (FrankIglesias) [inline wolox-books/src/views/Register.vue]: remove this semi colon

File: wolox-books/src/views/Register.vue [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    password: {
+      required,
+      validatePassword
+    },
+    email: {
+      required,
+      email
+    }
+  }
+};
```
Comment (FrankIglesias) [inline wolox-books/src/views/Register.vue]: same

File: wolox-books/src/views/Register.vue [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    .error(v-if="!$v.password.validatePassword")
+      | Password must have at least one number and one uppercase character
+    button
+      | {{$t('register.signUp')}}
+  a.button-login(href="/")
+    | {{$t('register.logIn')}}
+</template>
+
+<script>
+import { required, email, helpers } from "vuelidate/lib/validators";
```
Comment (FrankIglesias) [inline wolox-books/src/views/Register.vue]: smae

File: wolox-books/src/views/Register.vue [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+const validatePassword = helpers.regex(
+  "validatePassword",
+  /(?=.*[a-z])(?=.*[A-Z])(?=.*\d)/
+);
```
Comment (FrankIglesias) [inline wolox-books/src/views/Register.vue]: remove

File: wolox-books/src/views/Register.vue [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    onSubmit() {
+      const data = {
+        first_name: this.firstName,
+        last_name: this.lastName,
+        email: this.email,
+        password: this.password
+      };
+      console.log(data);
+      this.$v.$touch();
+      if (this.$v.$invalid) {
```
Comment (FrankIglesias) [inline wolox-books/src/views/Register.vue]: remove this curly braces

File: wolox-books/src/views/Register.vue [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  width: 300px;
+  align-self: center;
+}
+
+.label {
+  text-align: left;
+}
+
+.button-login {
+  width: 300px;
```
Comment (FrankIglesias) [inline wolox-books/src/views/Register.vue:91]: use width and max width (maybe this class can be a generic one)

File: wolox-books/src/views/Register.vue [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    },
+    email: {
+      required,
+      email
+    }
+  }
+};
+</script>
+
+<style lang="scss">
```
Comment (FrankIglesias) [inline wolox-books/src/views/Register.vue]: you'd like to have this classes scoped

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

File: wolox-books/src/views/Register.vue [hunk 8]
```diff
@@ -2,33 +2,45 @@
...
+    .error(v-if='!$v.password.required && submitStatus')
+      | Password is required
+    .error(v-if='!$v.password.validatePassword')
+      | Password must have at least one number and one uppercase character
     button
-      |{{$t('register.signUp')}}
-  a(href='/')
-    |{{$t('register.logIn')}}
+      | {{$t('register.signUp')}}
+  a.button-login(href='/')
```
Comment (mcallegari10) [inline wolox-books/src/views/Register.vue]: This won't work, use `router-link` component

**Review by mcallegari10 — APPROVED** [review]

**Review by mcallegari10 — APPROVED** [review]

**Pull request merged**